### ISSUE TITLE
Horizontal bands instead of images in random batches

### ISSUE THREAD

**Issue by jbohnslav** [NVIDIA/nvvl#86]
Hello, 

Thanks for all your great work! I'm working on a project wherein I save optical flow data as RGB images, then encode them in .mp4 files. I calculated flow with TV-L1, made it an image by putting dX in the G channel and dY in the R channel, then converted the resulting movie using ffmpeg according to your .readme:
` ffmpeg -y -i \'%s\' -map v:0 -c:v libx264 -crf 18 -pix_fmt yuv420p -g 8 -profile:v high %s' %(in_name, out_name)`

I initialize a dataset using `sequence_length` of 10 and `batch_size` of 64:

```
processing = {'input': nvvl.ProcessDesc(normalized=True)}
dataset = nvvl.VideoDataset(vidfiles, sequence_length, processing=processing, device_id=1, 
                           get_label=get_label)
loader = nvvl.VideoLoader(dataset, batch_size=64, shuffle=False)
```
I post-process the resulting tensors by subtracting 0.5 so that flow values are centered around zero. I also reshape the batches from `[N,T,C,H,W]` to `[N,T*2, H,W]`. That way the inputs to my network are a stack of optical flow images.

Some of the batches look like correct, if of course a bit quantized:
![image](https://user-images.githubusercontent.com/17788259/49535402-21284100-f892-11e8-8a35-46636207699d.png)

Others show nothing but horizontal bands:
![image](https://user-images.githubusercontent.com/17788259/49535431-33a27a80-f892-11e8-8dce-2a08a81863da.png)

Do you know what the issue is? It doesn't occur every N batches or every N sequences within a batch. It seems pretty random.

**Comment by jbohnslav** [comment]
Thanks for the comment! Following your advice, I'm now using the `dataset` class to retrieve individual sequences. Banding still appears sometimes, so it is reproducible without the complexity of the `loader`. 

The most confusing part is: I've realized that the same exact batch is sometimes banded! Here is the zeroth batch of the zeroth file in my dataset. The code is: 
`first_batch = dataset[0]`
`batch = first_batch['input'].detach().cpu().numpy()`
`# plot_batch(...)`

![image](https://user-images.githubusercontent.com/17788259/49611191-cc0e2d00-f96e-11e8-9e9c-34cc69b4f4be.png)
And here it is again, the same exact code block run again: 
![image](https://user-images.githubusercontent.com/17788259/49611279-0d064180-f96f-11e8-95c8-a04d1e1a61bc.png)

This allowed me to spot the pattern: the image values are the left-most column of data, repeated! Here is the line of code: 
`first_column_only = np.repeat(batch[:,:,:,0:1], batch.shape[3], axis=3)`
`# plot_batch(first_column_only)`

![image](https://user-images.githubusercontent.com/17788259/49612849-eac2f280-f973-11e8-943f-ae03b44504e2.png)

Comparing to the random instances that it broke:
`np.array_equal(messed_up_batch, first_column_only) # prints True`

Do you have any idea what could be causing the first column of data to be repeated? It's very strange that the same block of code (`batch = dataset[0]['input']`) can randomly be correct or incorrect.

**Comment by jaredcasper** [comment]
ok, actually its in both... looks like for some reason the default for random_flip in ProcessDesc got set to true, and the calculation for doing a flip when not scaling is broken. In your case if you don't want the random flip, you can change the default, (line 72 of dataset.py) or just set random_flip=False, and see if that fixes it?

**Comment by jaredcasper** [comment]
Just pushed a change that will hopefully fix this, please let me know if it does.

**Comment by jbohnslav** [comment]
Just installed the new package. Thanks for your work!

All banding is removed when I use the following processing pipeline: 
`processing = {'input': nvvl.ProcessDesc(normalized=True, random_flip=False)}`

However, with the following processing I still see banding:
`processing = {'input': nvvl.ProcessDesc(normalized=True, random_flip=True)}`

I tried a bunch of different parameters. When I randomly flip and also rescale (i.e. with `scale_width` or `scale_height != 0`) there is no banding. It seems like there is something the matter still with the unscaled but randomly flipped case.
